heartbeat/monitors/factory.go

```diff
@@ -232,6 +258,27 @@ func preProcessors(info beat.Info, settings publishSettings, monitorType string)
 	// Always set event.dataset
 	procs.AddProcessor(actions.NewAddFields(mapstr.M{"event": mapstr.M{"dataset": dataset}}, true, true))
 
+	// If we have a location to add, use the add_observer_metadata processor
+	if location != nil {
+		var geoM mapstr.M
+
+		geoM, err := util.GeoConfigToMap(location.Geo)
+		if err != nil {
+			geoErrOnce.Do(func() {
```

Comment:
Neat use of singleton ❤️

Reply:
@andrewc a bit of a late question, sorry about that. Isn't this statement assuming that there'll be only one location across all available monitors? In that case, would it be a fair assumption to make?